src/coreclr/tools/Common/TypeSystem/IL/CoreRTILProvider.cs

```diff
@@ -114,6 +114,28 @@ private MethodIL TryGetPerInstantiationIntrinsicMethodIL(MethodDesc method)
 
             switch (owningType.Name)
             {
+                case "Activator":
+                    {
+                        TypeSystemContext context = owningType.Context;
+                        if (methodName == "CreateInstance" && method.Signature.Length == 0 && method.HasInstantiation
+                            && method.Instantiation[0] is TypeDesc activatedType
+                            && activatedType != context.UniversalCanonType
+                            && activatedType.IsValueType
+                            && activatedType.GetParameterlessConstructor() == null)
+                        {
+                            ILEmitter emit = new ILEmitter();
+                            ILCodeStream codeStream = emit.NewCodeStream();
+
+                            var t = emit.NewLocal(context.GetSignatureVariable(0, method: true));
+                            codeStream.EmitLdLoca(t);
+                            codeStream.Emit(ILOpcode.initobj, emit.NewToken(context.GetSignatureVariable(0, method: true)));
```

Comment:
I guess crossgen2 would choke on this, and that's why you choose to not have this in the ILProvider base type?

Reply:
Yep. The strategy is the same as crossgen2 but crossgen2 can't make up new tokens. I expect using the exact same strategy as crossgen2 (returning a different `EcmaMethodIL`) would lead to problems. The crossgen2 strategy is already leaking out in places and it's annoying (https://github.com/dotnet/runtime/blob/bcc7e0e0ce1758322015675356efc121b211e312/src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs#L1238).